packages/sdk/src/sync-agent/user/models/metadata/displayName.ts

```diff
@@ -0,0 +1,118 @@
+import { check, dlogger } from '@river-build/dlog'
+import { LoadPriority, type Identifiable, type Store } from '../../../../store/store'
+import {
+    PersistedObservable,
+    persistedObservable,
+} from '../../../../observable/persistedObservable'
+import type { RiverConnection } from '../../../river-connection/riverConnection'
+import type { Client } from '../../../../client'
+import { isDefined } from '../../../../check'
+import type { IStreamStateView } from '../../../../streamStateView'
+import { make_MemberPayload_DisplayName } from '../../../../types'
+const logger = dlogger('csb:userSettings')
+
+type DisplayNameModel = {
+    isEncrypted: boolean
+    displayName: string
+}
+
+export interface UserMetadata_DisplayNameModel extends Identifiable {
+    id: string
+    streamId: string
+    initialized: boolean
+    displayNames: Map<string, DisplayNameModel | undefined>
+}
+
+@persistedObservable({ tableName: 'UserMetadata_DisplayName' })
+export class UserMetadata_DisplayName extends PersistedObservable<UserMetadata_DisplayNameModel> {
+    constructor(
+        userId: string,
+        streamId: string,
+        store: Store,
+        private riverConnection: RiverConnection,
+    ) {
+        super(
+            { id: userId, streamId, initialized: false, displayNames: new Map() },
```

Comment:
note - we want to use this in a member object which will exist on every stream, because the same user id can belong to multiple streams, the id should be `${streamId}_${userId}`